src/Mod/PartDesign/Gui/TaskLinearPatternParameters.cpp

```diff
@@ -255,44 +219,29 @@ void TaskLinearPatternParameters::kickUpdateViewTimer() const
     updateViewTimer->start();
 }
 
-void TaskLinearPatternParameters::addObject(App::DocumentObject* obj)
-{
-    QString label = QString::fromUtf8(obj->Label.getValue());
-    QString objectName = QString::fromLatin1(obj->getNameInDocument());
-
-    QListWidgetItem* item = new QListWidgetItem();
-    item->setText(label);
-    item->setData(Qt::UserRole, objectName);
-    ui->listWidgetFeatures->addItem(item);
-}
-
-void TaskLinearPatternParameters::removeObject(App::DocumentObject* obj)
-{
-    QString label = QString::fromUtf8(obj->Label.getValue());
-    removeItemFromListWidget(ui->listWidgetFeatures, label);
-}
-
 void TaskLinearPatternParameters::onSelectionChanged(const Gui::SelectionChanges& msg)
 {
-    if (selectionMode != none && msg.Type == Gui::SelectionChanges::AddSelection) {
+    if (selectionMode != SelectionMode::None && msg.Type == Gui::SelectionChanges::AddSelection) {
         if (originalSelected(msg)) {
             exitSelectionMode();
         }
-        else if (selectionMode == reference) {
+        else if (selectionMode == SelectionMode::Reference) {
             // TODO check if this works correctly (2015-09-01, Fat-Zer)

```

Comment:
While we're touching all this code, can you either actually check this, or tweak this line to read: 
```suggestion
            // TODO: check if this works correctly (2015-09-01, Fat-Zer)
```
(I think most auto-generation tools use the colon to detect these TODO statements).

Reply:
The code was different between the patterns and PolarPattern was missing some class checks.
It seems to work now correctly.